### PR TITLE
Use yarn lock

### PR DESCRIPTION
**Pull request by bourdakos1**
This PR removes the yarn.lock file from the gitignore in order to reduce the occurrence of build-time errors.

closes: #1544
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by lresende** [comment]
Should you include an initial yarn lock with the pr? And document what to do and when to update it.

**Comment by lresende** [comment]
You were quick, looks like yarn.lock has already beein added.

**Comment by bourdakos1** [comment]
>  And document what to do and when to update it.

@lresende nothing should really change from a development perspective

**Comment by bourdakos1** [comment]
Actually, maybe we should document merge conflicts? Where should that go?

If there is a merge conflict with the `yarn.lock` you shouldn't manually resolve it, but instead just run `yarn` or `yarn install` and it will resolve the conflict for you

**Comment by lresende** [comment]
> Actually, maybe we should document merge conflicts? Where should that go?
> 
> If there is a merge conflict with the `yarn.lock` you shouldn't manually resolve it, but instead just run `yarn` or `yarn install` and it will resolve the conflict for you

Maybe on the developer workflow doc?

**Comment by lresende** [comment]
Also, the release script build, should it actually use the yarn lock file or be a fresh build from scratch (e.g. remove the yarn lock file first)?

**Comment by bourdakos1** [comment]
> Also, the release script build, should it actually use the yarn lock file or be a fresh build from scratch (e.g. remove the yarn lock file first)?

We should keep the `yarn.lock` in the release script. We should never remove the `yarn.lock` for anything (The only exception is the possibility of removing it for the integration tests).

**Comment by lresende** [comment]
Overall LGTM, but I would like to get an approval from others on the frontend side (@karlaspuldaro, @marthacryan, @ajbozarth)

**Comment by bourdakos1** [comment]
@lresende I just added a section to the docs "Contributing back to Elyra" with a note about merge conflicts with the `yarn.lock`. Does that all look okay?

**Review by marthacryan — APPROVED** [review]
Haven't tried locally but LGTM

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -75,6 +75,8 @@ jobs:
     - name: Build
       run: |
         export PATH=~/.npm-global/bin:$PATH
+        # make sure nothing was manually added to the package.json
+        yarn install --frozen-lockfile
```
Comment (lresende) [inline .github/workflows/build.yml:79]: `make install` already calls yarn install... could you please update the makefile instead

**Comment by lresende** [comment]
> @lresende I just added a section to the docs "Contributing back to Elyra" with a note about merge conflicts with the `yarn.lock`. Does that all look okay?

We have a generic [contributor guidelines](https://github.com/elyra-ai/community/blob/master/contributing.md#before-creating-a-pull-request) that is linked from every `CONTRIBUTING.MD` in all repositories. Should this actually go there, together with the section that talks about resolving conflicts?

@ptitzler Thoughts here on documentation location?

File: .github/workflows/build.yml (hunk 1 above)
Comment (bourdakos1) [reply .github/workflows/build.yml:79]: `yarn install --frozen-lockfile` is something that should only need to be run in ci. It is basically just a check to make sure that a contributor didn't use `npm install` to add a package, or manually add something to the package.json and never ran `yarn install` to install things

Comment (bourdakos1) [reply .github/workflows/build.yml:79]: Adding `yarn install --frozen-lockfile` to `make install` would throw an error if you tried to add something to a package.json and follow up with a `make install` (something that should be okay to do). Without `yarn install --frozen-lockfile` in `make install`, adding something to the package.json and then running `make install`, would just install the packages and update the `yarn.lock`, which is normally the expected behavior

Comment (bourdakos1) [reply .github/workflows/build.yml:79]: The npm equivalent of this is `npm ci`: https://docs.npmjs.com/cli/v7/commands/npm-ci

**Comment by bourdakos1** [comment]
Oh yes, this looks like a better place to put it, I'll remove my doc changes and open a PR there instead

**Review by ajbozarth — APPROVED** [review]
like Martha I didn't test this locally, but afaik it lgtm

**Review by lresende — APPROVED** [review]

**Pull request merged**